### PR TITLE
Added check on dining_time to adhere kitchen opening time

### PR DESCRIPTION
**Pull request by DutcherNL**
Limited dining time in forms to the kitchen use times

### CODE REVIEW

File: Dining/forms.py [hunk 1]
```diff
@@ -53,6 +54,15 @@ def __init__(self, user, date, *args, **kwargs):
             self.fields['association'].initial = available[0].pk
             self.fields['association'].disabled = True
 
+    def clean(self, *args, **kwargs):
+        cleaned_data = super(CreateSlotForm, self).clean(*args, **kwargs)
+        # Check if the time is within time limits
+        serve_time = cleaned_data['serve_time']
+        if serve_time < settings.KITCHEN_USE_START_TIME:
+            self.add_error('serve_time', _("Kitchen can't be used this early"))
```
Comment (mhvis) [inline Dining/forms.py]: Should be `raise ValidationError()` I think.

File: Dining/forms.py [hunk 2]
```diff
@@ -53,6 +54,15 @@ def __init__(self, user, date, *args, **kwargs):
...
+        if serve_time > settings.KITCHEN_USE_END_TIME:
+            self.add_error('serve_time', _("Kitchen can't be used this late"))
```
Comment (mhvis) [inline Dining/forms.py]: Needs `return cleaned_data` at the end to correctly override `clean()`

File: Dining/forms.py [hunk 3]
```diff
@@ -76,6 +86,15 @@ class Meta:
...
         fields = ['serve_time', 'min_diners', 'max_diners', 'sign_up_deadline', 'purchaser']
 
+    def clean(self, *args, **kwargs):
+        cleaned_data = super(DiningInfoForm, self).clean(*args, **kwargs)
+        # Check if the time is within time limits
+        serve_time = cleaned_data['serve_time']
+        if serve_time < settings.KITCHEN_USE_START_TIME:
+            self.add_error('serve_time', _("Kitchen can't be used this early"))
+        if serve_time > settings.KITCHEN_USE_END_TIME:
+            self.add_error('serve_time', _("Kitchen can't be used this late"))
```
Comment (mhvis) [inline Dining/forms.py]: Needs `return cleaned_data`.

Also duplicate code, maybe better to put in model clean or MixIn or superclass.

File: Dining/forms.py (hunk 1 above)
Comment (DutcherNL) [reply Dining/forms.py]: Nope, this makes the error appear at the specefic field instead of the full form.

File: Dining/forms.py (hunk 3 above)
Comment (DutcherNL) [reply Dining/forms.py]: Used Mixin, that was a good idea.

**Pull request merged**